### PR TITLE
updated app.js

### PR DESCRIPTION
**Pull request by p-syche**
fixed a few issues, more details in the code

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
...
-	const jsonValue = await AsyncStorage.getItem('@tasks')
-	 return jsonValue != null ? JSON.parse(jsonValue) : null;
-        } catch(err) {
-		console.log(err);
+      const jsonValue = await AsyncStorage.getItem('@tasks')
+      
+      console.log('inside get promise', jsonValue, JSON.parse(jsonValue))
+      return jsonValue != null ? JSON.parse(jsonValue) : null;
+    } catch(err) {
+      console.log(err);
```
Comment (p-syche) [inline App.js:28]: I moved the `storeData` and `getData` functions to the very top of the file to make sure we are able to call them later on. I left a `console.log` by accident, but I recommend checking your data every step of the way.

File: App.js [hunk 2]
```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
...
-	}
-			    
+}
+  
+ const addTodo = () => {
+  let id = todos[todos.length - 1]?.id || 0 ;
+   let thisId = id * 1 + 1 + "";
+   const updatedToDoArray = [...todos, { id: thisId, text }];
+  setTodos(updatedToDoArray);
+  storeData(updatedToDoArray)
```
Comment (p-syche) [inline App.js:37]: I moved `storeData` to the function which adds a todo. That way every time someone adds a todo the array of todos stored will be updated. You had your storeData in a `useEffect` hook, which had an empty dependency array, which means the function run ONLY when the component was mounted. I invite you to read more about the `useEffect` hook in the react docs: https://reactjs.org/docs/hooks-reference.html#useeffect

File: App.js (hunk 1 above)
Comment (ViNoS-ab) [reply App.js:28]: ok thanks for your help , i really appreciate it

File: App.js [hunk 3]
```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
...
+      console.log(err);
 	  }
-	}
-			    
+}
+  
+ const addTodo = () => {
+  let id = todos[todos.length - 1]?.id || 0 ;
+   let thisId = id * 1 + 1 + "";
+   const updatedToDoArray = [...todos, { id: thisId, text }];
```
Comment (p-syche) [inline App.js:35]: As you can see I created a const value with the updated Array of todos. You can't count on running this filter function inside `setTodos` and have it updated in time to pass `todos` to `storeData`. You have to have one source of truth: your `updatedToDoArray` is the source of truth for your useState hook AND your storeData function.

File: App.js [hunk 4]
```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
...
+  storeData(updatedToDoArray)
+  onChangeText("");
+ }
+ 
 
-const remove = id => { 
-setTodos(() =>todos.filter(item => item.id !== id )) ; 
+  const remove = id => { 
+    const updatedToDoArray = todos.filter(item => item.id !== id);
+    setTodos(updatedToDoArray);
```
Comment (p-syche) [inline App.js:44]: You wrote the filter function INSIDE the useState hook. Your useState hook is set to hang on to an Array. You can pass an Array to it, you really shouldn't pass a function.

File: App.js [hunk 5]
```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
...
 
-useEffect(() => {
- storeData(todos);
-}, [todos] 
-)
+useEffect(() => { 
+  getData().then(storedData => { 
+    if (storedData.length > 0) {
+      setTodos(storedData)
+  console.log('tell me in the get', storedData);
```
Comment (p-syche) [inline App.js:51]: Always check what sort of data you are passing.

File: App.js [hunk 6]
```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
...
+    const updatedToDoArray = todos.filter(item => item.id !== id);
+    setTodos(updatedToDoArray);
 }
 
-useEffect(() => {
- storeData(todos);
-}, [todos] 
-)
+useEffect(() => { 
+  getData().then(storedData => {
```
Comment (p-syche) [inline App.js:48]: This is super important! 
`getData` is a Promise. It doesn't return a value like a regular JS function, you can't do 

`const something = getData()`

I mean, you can... but it won't work. You have to use `.then()` with Promises.
You can read more about Promises here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

**Pull request merged**